cumulus/client/consensus/aura/src/collators/slot_based/tests.rs

```diff
@@ -104,6 +109,39 @@ async fn offset_test_too_long() {
 	assert!(result.is_err());
 }
 
+#[tokio::test]
+async fn offset_returns_none_when_best_header_contains_epoch_change() {
```

Comment:
Nice tests!

One nitpick is that you could use `rstest`, example here: https://github.com/paritytech/polkadot-sdk/blob/9fb3757b11e7d8bf23e57a94a3a1639d1ad8d0bc/cumulus/pallets/parachain-system/src/descendant_validation.rs#L289-L289

I like that the parameterization makes the file a bit shorter. But optional, do as you like here.